networkx/classes/function.py

```diff
@@ -346,40 +368,59 @@ def get_node_attributes(G, name):
 
 
 def set_edge_attributes(G, name, values):
-    """Set edge attributes from dictionary of edge tuples and values.
+    """Sets node attributes from a given value or dictionary of values.
 
     Parameters
     ----------
     G : NetworkX Graph
 
     name : string
-       Attribute name
+       Name of the edge attribute to set.
 
     values : dict
-       Dictionary of attribute values keyed by edge (tuple). For multigraphs,
-       the keys tuples must be of the form (u, v, key). For non-multigraphs,
-       the keys must be tuples of the form (u, v). If `values` is not a
-       dictionary, then it is treated as a single attribute value that is then
-       applied to every edge in `G`.
+
+       Dictionary of attribute values keyed by edge (tuple). For
+       multigraphs, the tuples must be of the form ``(u, v, key)``. For
+       non-multigraphs, the keys must be tuples of the form ``(u,
+       v)``.
+
+       If ``values`` is not a dictionary, then it is treated as a single
+       attribute value that is then applied to every edge in ``G``. This
+       means that if you provide a mutable object, like a list, updates
+       to that object will be reflected in the edge attribute for each
+       edge.
 
     Examples
     --------
-    >>> G = nx.path_graph(3)
-    >>> bb = nx.edge_betweenness_centrality(G, normalized=False)
-    >>> nx.set_edge_attributes(G, 'betweenness', bb)
-    >>> G[1][2]['betweenness']
-    2.0
+    After computing some property of the nodes of a graph, you may want
+    to assign a node attribute to store the value of that property for
+    each node::
```

Comment:
It's not necessary, I use it as a safeguard against future changes to the documentation rendering. The double colon + indentation is the plain-old Sphinx default way of denoting an upcoming code block, and it renders as a single colon before the code block in the rendered documentation. Numpydoc is smarter, so it doesn't need the double colons. I only do this for backward-compatibility with Sphinx, but I'm happy to stop doing that if you prefer.